### PR TITLE
Makes Ants glow, puts a min on ant screaming & shoe permeability, & other ant-related things.

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
I found out how emissives work and my first thought was "damn ants should glow that would look sick"
So now they do.

https://user-images.githubusercontent.com/66052067/153104418-95157a25-3c85-4859-b627-dd3e0890014a.mp4

Also, having less than 5u ants in your body will make you not scream, so 0.0001u ants will no longer have that tiny chance of making someone scream for their life.

If an ant pile has a max damage value less than 1, then they won't be able to bite through your shoes. This is the same threshold as the second tier ant icon.

Makes the giant ant a hostile mob with the neutral faction, meaning they will attack anything **not** in the neutral faction.

Finally, cleans up some messy code by moving the things you can scream to a list variable, as well as moving a sound file path so you can finally ctrl+click the caltrop path if you wanted to.

## Why It's Good For The Game

https://user-images.githubusercontent.com/66052067/153104470-b22811be-8fff-46ec-b509-3d0e81908cb3.mp4

Makes ant-based food more appetizing since you won't be screaming when you eat it.
People (me included) found the fact that stepping on ants meant 0.1 damage at random moments annoying, so now only massive amounts of ants will be able to bite through your shoes. The main downside of decomposition still persists, that being that your food gets destroyed.

## Changelog

:cl: Wallem
qol: Adds a minimum threshold to ingested ant screaming.
qol: Ants are no longer lavaproof.
balance: Adds a minimum threshold to ant piles biting through shoes.
balance: Giant ants are now peaceful hostile mobs. This means they will attack anything that does not have the "neutral" factiontype. To compensate, the price to make one has been raised.
imageadd: It appears the invasive "space ant" has gotten into our formerly secret uranium reserves and as such have mutated slightly, gaining a slight bioluminescence.
/:cl:

### CODE REVIEW

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -1001,6 +1001,15 @@
 	processing_speed = STATUS_EFFECT_NORMAL_PROCESS
 	/// Will act as the main timer as well as changing how much damage the ants do.
 	var/ants_remaining = 0
+	/// Common phrases people covered in ants scream
+	var/list/ant_debuff_speech = list(
```
Comment (MrMelbert) [inline code/datums/status_effects/debuffs.dm:1005]:
```suggestion
	var/static/list/ant_debuff_speech = list(
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -2743,19 +2743,29 @@
 	var/ant_damage = 0
 	/// Tells the debuff how many ants we are being covered with.
 	var/amount_left = 0
+	/// List of possible common statements to scream when eating ants
+	var/list/ant_screams = list(
```
Comment (MrMelbert) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2747]:
```suggestion
	var/static/list/ant_screams = list(
```
Normally, if you've got a long-ish list of strings or whatevers that you don't override with subtypes or mutate in some way, you can use a static list for performance.

Comment (Wallemations) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:2747]: Never knew that, thanks

Comment (LemonInTheDark) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: To add to this, this is very close to the usecase for string files. Not very long tho

Comment (Wallemations) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: I haven't been planning on making it longer myself, though I don't have the slightest clue if anyone else would want to add to it later.

**Comment by ATH1909** [comment]
> Also, having less than 5u ants in your body will make you not scream, so 0.0001u ants will no longer have that tiny chance of making someone scream for their life.

why would you not scream even if you had ""only"" 0.0001u of a stream of ants in your bloodstream

**Comment by Wallemations** [comment]
> > Also, having less than 5u ants in your body will make you not scream, so 0.0001u ants will no longer have that tiny chance of making someone scream for their life.
> 
> why would you not scream if you had ""only"" 0.0001u of a stream of ants in your bloodstream

Cause with an amount like that I'd say you just swallowed an ant's leg on its own lmao

**Comment by Wallemations** [comment]
I just remembered factions are a flag, so I'm gonna make the ant a hostile mob in the centcomm faction similar to how the HOS' pet carp is. If the tiny ones bite then the pet might as well bite too.

**Comment by Wallemations** [comment]
Nearly forgot to also add lava tiles to the decomposition typecache too

**Comment by Wallemations** [comment]
It says that it failed to compile the maps but I didn't do anything to edit the maps??
That isn't gonna be a problem is it?

**Review by Twaticus — APPROVED** [review]

**Review by MrMelbert — CHANGES_REQUESTED** [review]
```
_maps/RandomRuins/IceRuins/icemoon_surface_bughabitat.dmm:32:error: undefined type: /mob/living/simple_animal/ant
```
```
_maps/RandomRuins/IceRuins/icemoon_surface_bughabitat.dmm:34:error: unknown type
```
you repathed ants, but a map uses the old path still

**Comment by Wallemations** [comment]
> ```
> _maps/RandomRuins/IceRuins/icemoon_surface_bughabitat.dmm:32:error: undefined type: /mob/living/simple_animal/ant
> ```
> 
> ```
> _maps/RandomRuins/IceRuins/icemoon_surface_bughabitat.dmm:34:error: unknown type
> ```
> 
> you repathed ants, but a map uses the old path still

**_Somebody used ants on a map?_**

**Comment by Wallemations** [comment]
I made that one ant in there have `obj_damage = 0` so it can't break into the cockroach pen next to it to try and eat/kill them.

**Comment by Wallemations** [comment]
found out `obj_damage = 0` just makes it BETTER at destroying things, I fixed it for real this time.

**Comment by Wallemations** [comment]
wow somehow that ended in A DIFFERENT check fail.

**Comment by Wallemations** [comment]
ok just because the PR has been open for a while there's a map conflict oops

**Pull request merged**